apps/rebar/src/rebar_prv_completion.erl

```diff
@@ -0,0 +1,165 @@
+%% @doc Generates shell completion files based on available providers and their opts.
+%% @end
+-module(rebar_prv_completion).
+
+-behaviour(provider).
+
+-export([init/1,
+         do/1,
+         format_error/1]).
+
+-include_lib("providers/include/providers.hrl").
+-include("rebar.hrl").
+
+-define(PROVIDER, completion).
+-define(DEPS, [app_discovery]).
```

Comment:
I'm assuming we want `app_discovery` here to detect plugins? If so that's okay.

Reply:
Yup, this is for plugins